### PR TITLE
Pp 12421: Add Classes for Git and Docker Image resources

### PR DESCRIPTION
**Pull request by kerrr**
The structure of resource definitions is concourse is dependant on the type. We can add pkl classes to represent these. Here we are adding classes for the Git and DockerImage resources.

### CODE REVIEW

File: src/Resources.pkl [hunk 1]
```diff
@@ -0,0 +1,97 @@
+module concourse_pipeline.Resources
+
+import "./Pipeline.pkl"
+
+open class GitResource extends Pipeline.Resource{
+  type = "git"
+  icon = "github"
+  source: GitResourceConfig?
```
Comment (kbottla) [inline src/Resources.pkl:8]: Can `source` be mandatory?

File: src/Resources.pkl [hunk 2]
```diff
@@ -0,0 +1,97 @@
+module concourse_pipeline.Resources
+
+import "./Pipeline.pkl"
+
+open class GitResource extends Pipeline.Resource{
```
Comment (kbottla) [inline src/Resources.pkl:5]: Missing space here and in some places further. Worth formatting the file

```suggestion
open class GitResource extends Pipeline.Resource {
```

File: src/Resources.pkl [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+class GitResourceConfig{
+  uri: Uri
+  branch: String?
+  private_key: String?
+  private_key_user: String?
+  private_key_passphrase: String?
+  forward_agent: Boolean?
+  username: String?
```
Comment (kbottla) [inline src/Resources.pkl:18]: (minor) Could make username and password fields mandatory too so we don't end up getting rate limited when resource is used without these fields

File: src/Resources.pkl [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+open class DockerImageResource extends Pipeline.Resource {
+  type = "registry-image"
+  icon = "docker"
+  source: DockerImageResourceConfig?
+}
+
+class DockerImageResourceConfig{
+  repository: String
+  tag: String?
+  username: String?
```
Comment (kbottla) [inline src/Resources.pkl:74]: Same as Github username & password. Could make these mandatory

File: src/Resources.pkl (hunk 3 above)
Comment (kerrr) [reply src/Resources.pkl:18]: I think we want the classes here to reflect the resource type docs. We can then we can exend for our internal usage.

File: src/Resources.pkl (hunk 4 above)
Comment (kerrr) [reply src/Resources.pkl:74]: As above.

**Review by kbottla — APPROVED** [review]

**Pull request merged**